app/services/metrics/review_turnaround/per_user_project.rb

```diff
@@ -3,29 +3,57 @@ module ReviewTurnaround
     class PerUserProject < BaseService
       BATCH_SIZE = 500
 
+      def initialize(interval = nil)
+        @interval = interval
+      end
+
       def call
         process
       end
 
       private
 
       def process
-        filtered_reviews_ids.each_slice(BATCH_SIZE) do |batch|
-          Events::Review.eager_load(:review_request).find(batch).lazy.each do |review|
-            entity = find_user_project(review.owner, review.pull_request.project)
-            turnaround = calculate_turnaround(review)
+        ActiveRecord::Base.transaction do
```

Comment:
should the transaction be here or inside the loop? Asking because if it fails maybe we can keep the iterations that were processed correctly

Reply:
I need to keep `calculate_avg` inside the transaction because if something fails inside `calculate_avg` I would have Metrics with wrong data in the database, because I don't have the correct value of the Metric until `calculate_avg` happens.